internal/services/controller/worker.go

```diff
@@ -105,7 +58,21 @@ func RunController(
 		)
 		f.Start(ctrlCtx.Done())
 
-		// Run the controller. This is a blocking call.
+		// Loop the controller. This is a blocking call.
 		return ctrl.Run(ctrlCtx)
+	})
+}
+
+func repeatInContext(ctx context.Context, fn func(context.Context) error) error {
```

Comment:
Would suggest renaming to `repeatUntilContextClosed`. I think it would be more explicit on what it does.